File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/interfaces/interfaceWithInterfaceInheritance/expected/types/StoneFruit.java
package com.netflix.graphql.dgs.codegen.java.testcases.interfaces.interfaceWithInterfaceInheritance.expected.types;

import java.lang.Boolean;
import java.util.List;

public interface StoneFruit extends Fruit {
  List<Seed> getSeeds();

  void setSeeds(List<Seed> seeds);

  Boolean getFuzzy();

  void setFuzzy(Boolean fuzzy);
}

File: graphql-dgs-codegen-core/src/test/resources/kotlin2/interfaceWithInterfaceInheritance/expected/types/StoneFruit.kt
package kotlin2.interfaceWithInterfaceInheritance.expected.types

import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import kotlin.Boolean
import kotlin.collections.List

@JsonTypeInfo(
  use = JsonTypeInfo.Id.NAME,
  include = JsonTypeInfo.As.PROPERTY,
  property = "__typename",
)
public sealed interface StoneFruit : Fruit {
  public override val seeds: List<Seed?>?

  public val fuzzy: Boolean?
}
